traitsui/qt4/range_editor.py

```diff
@@ -719,14 +719,18 @@ def update_object(self):
 # -------------------------------------------------------------------------
 
 
-def SimpleEnumEditor(parent, factory, ui, object, name, description):
+def SimpleEnumEditor(
+    parent, factory, ui, object, name, description, format_func=None,
+    format_str='', invalid_trait_name=''
+):
```

Comment:
Perhaps we should use `**kwargs` here. I am not a fan of using `**kwargs` myself but this function is trying to imitate the instantiation of a subclass of `HasTraits`, so it makes sense for the function signature to match.

Reply:
I'm a bit worried that `**kwargs` might end up masking potential future bugs. I've added a couple of tests, so now similar problems of signatures not matching (if not using `kwargs`) should be caught early. And given that these "fake" editors are quite hidden, maybe it's better for things to break early rather than accepting everything quietly?

Reply:
If I was to re-design these factories, I would not use `**kwargs` for exact the reason that is too lenient. Back to what we have here...

These functions are trying to duck type a subclass of HasTraits, here:
https://github.com/enthought/traitsui/blob/ab90190dded4c70b2ba39e9323b6ef424c926dd4/traitsui/editors/range_editor.py#L237-L238

When it gets to this point:
https://github.com/enthought/traitsui/blob/0881435aed3d8390a6fd7cf795b2f6df679b9085/traitsui/editor_factory.py#L146
One assumes a subclass of `HasTraits`. The implicit interface there is therefore the signature of `(*args, **kwargs)`.  Not accepting `**kwargs` broke that implicit contract consistent across the code base.

Reply:
While we are at this, it would safe to add `*` before `factory`. This is a new feature of Python 3 that we can now use because Python 2 is dropped. Yay.
Only the `parent` is expected to be passed onto `init`:
https://github.com/enthought/traitsui/blob/0881435aed3d8390a6fd7cf795b2f6df679b9085/traitsui/editor_factory.py#L91-L95

Reply:
Except... I don't know if this `SimpleEnumEditor` is used externally outside of the `editor_factory`. Adding `*` would be backward incompatible. Peril of having all the modules and classes with public names... 
I am sorry. Please ignore the `*` suggestion. I will open a separate issue to confirm. That is less important than the bug fix.